test/testio.cpp

```diff
@@ -748,64 +749,72 @@ class TestIO : public TestFixture {
-    ASSERT_EQUALS(TEST_SCANF_ERR(FORMAT,FORMATSTR,TYPE), errout.str());    \
-    check(TEST_SCANF_CODE(FORMAT,TYPE), true, false, Settings::Win32A);    \
-    ASSERT_EQUALS(TEST_SCANF_ERR(FORMAT,FORMATSTR,TYPE), errout.str());    \
-    check(TEST_SCANF_CODE(FORMAT,TYPE), true, false, Settings::Win64);     \
-    ASSERT_EQUALS(TEST_SCANF_ERR(FORMAT,FORMATSTR,TYPE), errout.str());
-
-#define TEST_SCANF_WARN_AKA(FORMAT, FORMATSTR, TYPE, AKATYPE, AKATYPE_WIN64)        \
-    check(TEST_SCANF_CODE(FORMAT,TYPE), true, true, Settings::Unix32);              \
-    ASSERT_EQUALS(TEST_SCANF_ERR_AKA(FORMAT,FORMATSTR,TYPE,AKATYPE), errout.str()); \
-    check(TEST_SCANF_CODE(FORMAT,TYPE), true, true, Settings::Unix64);              \
-    ASSERT_EQUALS(TEST_SCANF_ERR_AKA(FORMAT,FORMATSTR,TYPE,AKATYPE), errout.str()); \
-    check(TEST_SCANF_CODE(FORMAT,TYPE), true, true, Settings::Win32A);              \
-    ASSERT_EQUALS(TEST_SCANF_ERR_AKA(FORMAT,FORMATSTR,TYPE,AKATYPE), errout.str()); \
-    check(TEST_SCANF_CODE(FORMAT,TYPE), true, true, Settings::Win64);               \
-    ASSERT_EQUALS(TEST_SCANF_ERR_AKA(FORMAT,FORMATSTR,TYPE,AKATYPE_WIN64), errout.str());
-
-#define TEST_SCANF_WARN_AKA_WIN64(FORMAT, FORMATSTR, TYPE, AKATYPE_WIN64)           \
-    check(TEST_SCANF_CODE(FORMAT,TYPE), true, true, Settings::Unix32);              \
-    ASSERT_EQUALS("", errout.str());                                                \
-    check(TEST_SCANF_CODE(FORMAT,TYPE), true, true, Settings::Unix64);              \
-    ASSERT_EQUALS("", errout.str());                                                \
-    check(TEST_SCANF_CODE(FORMAT,TYPE), true, true, Settings::Win32A);              \
-    ASSERT_EQUALS("", errout.str());                                                \
-    check(TEST_SCANF_CODE(FORMAT,TYPE), true, true, Settings::Win64);               \
-    ASSERT_EQUALS(TEST_SCANF_ERR_AKA(FORMAT,FORMATSTR,TYPE,AKATYPE_WIN64), errout.str());
-
-#define TEST_SCANF_WARN_AKA_WIN32(FORMAT, FORMATSTR, TYPE, AKATYPE)                 \
-    check(TEST_SCANF_CODE(FORMAT,TYPE), true, true, Settings::Unix32);              \
-    ASSERT_EQUALS(TEST_SCANF_ERR_AKA(FORMAT,FORMATSTR,TYPE,AKATYPE), errout.str()); \
-    check(TEST_SCANF_CODE(FORMAT,TYPE), true, true, Settings::Unix64);              \
-    ASSERT_EQUALS(TEST_SCANF_ERR_AKA(FORMAT,FORMATSTR,TYPE,AKATYPE), errout.str()); \
-    check(TEST_SCANF_CODE(FORMAT,TYPE), true, true, Settings::Win32A);              \
-    ASSERT_EQUALS(TEST_SCANF_ERR_AKA(FORMAT,FORMATSTR,TYPE,AKATYPE), errout.str()); \
-    check(TEST_SCANF_CODE(FORMAT,TYPE), true, true, Settings::Win64);               \
-    ASSERT_EQUALS("", errout.str());
+    std::string testScanfErr(const std::string & format, const std::string & formatStr, const std::string & type) {
+        return std::string() + "[test.cpp:1]: (warning) " + format + " in format string (no. 1) requires '" + formatStr + " *' but the argument type is '" + type + " *'.\n";
```

Comment:
the std::string() at the start is redundant.

basically I like functions better than macros. however it is important to have quick tests since I run them very frequently. Do you have some timings with macros/functions? The macros will add the strings at compile time so how does this change affect the compilation time for test/testio.cpp? On the other hand I believe the macros should run faster, how is execution time affected for TestIO?